sdks/python/apache_beam/transforms/window.py

```diff
@@ -392,9 +392,11 @@ def __init__(self, size, period, offset=0):
 
   def assign(self, context):
     timestamp = context.timestamp
-    start = timestamp - (timestamp - self.offset) % self.period
-    return [IntervalWindow(Timestamp.of(s), Timestamp.of(s) + self.size)
-            for s in range(start, start - self.size, -self.period)]
+    start = timestamp - ((timestamp - self.offset) % self.period)
+    return [
+        IntervalWindow(Timestamp(micros=s), Timestamp(micros=s) + self.size)
```

Comment:
We could, that does change the return type from seconds -> microseconds. Do you know if there is any other place that depends on it, I wasn't sure about that part.